File: 0069-sqrtx/0069-sqrtx.js
/**
 * @param {number} x
 * @return {number}
 */
const mySqrt = x => {
    let left = 0, right = x;
    let root = -1;
    while (left <= right) {
        let mid = left + Math.floor((right - left) / 2);
        if (mid * mid === x) return mid;
        else if (mid * mid < x) left = mid + 1;
        else {
            root = mid;
            right = mid - 1;
        }
    }
    return root - 1;
};

File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left, right = 0, x
        root = 0
        while left <= right:
            mid = left + (right - left) // 2
            if mid * mid <= x:
                root = mid
                left = mid + 1
            else:
                right = mid - 1
        return root